app/common/collections/runner.py

```diff
@@ -0,0 +1,203 @@
+# todo: propose moving common/helper/collections.py to common/collections/submission.py
+from typing import TYPE_CHECKING, Callable, ClassVar, Optional
+from uuid import UUID
+
+from flask import request
+
+from app.common.collections.forms import build_question_form
+from app.common.data.types import FormRunnerState, SubmissionStatusEnum
+from app.common.forms import GenericSubmitForm
+from app.common.helpers.collections import SubmissionHelper
+from app.developers.forms import CheckYourAnswersForm
+from app.extensions import notification_service
+
+if TYPE_CHECKING:
+    from app.common.collections.forms import DynamicQuestionForm
+    from app.common.data.models import Form, Question
+    from app.common.data.models_user import User
+
+runner_url_map = dict[
+    FormRunnerState, Callable[["FormRunner", Optional["Question"], Optional["Form"], Optional[FormRunnerState]], str]
+]
+
+
+class FormRunner:
+    """Responsible for backing form runner pages, consistently takes care of:
+        - managing routing backwards and forwards
+        - integrity checks
+        - setting up forms
+
+    This allows us to implement the form runner in different domain environments consistently."""
+
+    url_map: ClassVar[runner_url_map] = {}
+    question: Optional["Question"] = None
+
+    def __init__(self, submission: SubmissionHelper):
+        self.submission = submission
```

Comment:
🤏  small wonder if we should make this a semi-private thing and then proxy the important things; could go either way.

```suggestion
        self._submission = submission
```

Reply:
I think this is a good idea but I've used the submission in a couple of places (views, templates) etc. so will do clean this up as a separate PR